Sage/carbon
# chore(link): remove classic theme - FE-3113

removal of classic theme from the link component. Story has been refactored as CSF. Redundant
documents have been deleted and replaced with an MDX example.

BREAKING CHANGE: Classic styling has been removed from the link the component. The link component no
longer supports the classic theme.

fixes FE-3113

### Proposed behaviour
- Remove `classic` theme
- Delete redundant documentation
- Refactor story to be CSF
- Create an MDX example

### Current behaviour
- Component currently supports `classic` theme
- Documentation is inline with our previous standards
- Story is in the deprecated `storiesOf` format

### Checklist



### Additional context
N/A

### Testing instructions
- `git checkout FE-3113-link_remove_classic`
- `npm start` to start Storybook
- Navigate to Link story in Storybook
- Component should not have any classic theme
- Example of component should be in MDX format
